api-server/rockcraft.yaml

```diff
@@ -19,7 +19,6 @@ services:
 
 
 package-repositories:
-  # Required for build-packages=python3.7
   - type: apt
     ppa: deadsnakes/ppa
     priority: always
```

Comment:
If we are using py3.8, do we still need this part?

Reply:
this is required to use a different python interpreter version from the build base, see https://documentation.ubuntu.com/rockcraft/en/latest/reference/plugins/python_plugin/#dependencies

we have to use 3.8, while the default for base (22.04) is 3.10

let me add a comment with the link to rockcraft docs

Reply:
without adding the ppa package repo, I was getting this error:
```
2024-11-20 12:42:33.111 :: 2024-11-20 12:42:32.511 Marking python3.8 (and its dependencies) to be fetched
2024-11-20 12:42:33.111 :: 2024-11-20 12:42:32.511 Cannot find package listed in 'build-packages': python3.8
2024-11-20 12:42:33.111 :: 2024-11-20 12:42:32.517 Traceback (most recent call last):
2024-11-20 12:42:33.111 :: 2024-11-20 12:42:32.517   File "/snap/rockcraft/2218/lib/python3.10/site-packages/craft_parts/packages/deb.py", line 508, in _get_packages_marked_for_installation
2024-11-20 12:42:33.111 :: 2024-11-20 12:42:32.517     apt_cache.mark_packages(set(package_names))
2024-11-20 12:42:33.111 :: 2024-11-20 12:42:32.517   File "/snap/rockcraft/2218/lib/python3.10/site-packages/craft_parts/packages/apt_cache.py", line 307, in mark_packages
2024-11-20 12:42:33.111 :: 2024-11-20 12:42:32.517     raise errors.PackageNotFound(name_arch)
2024-11-20 12:42:33.111 :: 2024-11-20 12:42:32.517 craft_parts.packages.errors.PackageNotFound: Package not found: python3.8.
```

Reply:
Awesome, the comment makes it so much better.